src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -413,10 +415,13 @@ private static void OnDocEnded(C4Replicator* repl, bool pushing, IntPtr numDocs,
                 return;
             }
 
+            var replicator = GCHandle.FromIntPtr((IntPtr)context).Target as Replicator;
+
             var replicatedDocumentsContainConflict = new List<ReplicatedDocument>();
             var documentReplications = new List<ReplicatedDocument>();
             for (int i = 0; i < (int) numDocs; i++) {
                 var current = docs[i];
+                //var collection = replicator.Config.Database.GetCollection(current->collectionName.CreateString());
                 if (!pushing && current->error.domain == C4ErrorDomain.LiteCoreDomain &&
                     current->error.code == (int) C4ErrorCode.Conflict) {
                     replicatedDocumentsContainConflict.Add(new ReplicatedDocument(current->docID.CreateString() ?? "",
```

Comment:
About the question of setting Collection = null in ReplicatedDocument constructor, the collection is supposedly to be created and pass in ReplicatedDocument (internal and only is used here) as parameter. The old constructor will be removed and setting Collection = null is temporarily because Database.GetCollection wasn't implemented yet, and I still want to existing tests to pass until the new implementation goes in...

Reply:
If it is temporary then ok